File: 01.03.URL化/01.03.cpp
class Solution {
public:
    string replaceSpaces(string S, int length) {
       int j = S.size() - 1;
        for(int i = length - 1; i >= 0; i--){
            if(S[i] == ' '){
                S[j--] = '0';
                S[j--] = '2';
                S[j--] = '%';
            } else{
                S[j--] = S[i]; 
            }
        }
        return S.substr(j + 1);
    }
};

File: 01.03.URL化/01.03.rs
impl Solution {
    pub fn replace_spaces(s: String, length: i32) -> String {
        let mut res = String::new();
        let mut ss = s.clone();
        ss.truncate(length as usize);

        for c in ss.chars() {
            if c == ' ' {
                res.push_str("%20");
            } else {
                res.push(c);
            }
        }
        res
    }
}
